zerver/lib/url_encoding.py

```diff
@@ -20,14 +21,16 @@ def encode_stream(stream_id: int, stream_name: str) -> str:
     return str(stream_id) + "-" + hash_util_encode(stream_name)
 
 
-def personal_narrow_url(realm: Realm, sender: UserProfile) -> str:
+def personal_narrow_url(*, realm: Realm, sender: UserProfile) -> str:
     base_url = f"{realm.uri}/#narrow/dm/"
     encoded_user_name = re2.sub(r'[ "%\/<>`\p{C}]+', "-", sender.full_name)
     pm_slug = str(sender.id) + "-" + encoded_user_name
     return base_url + pm_slug
 
 
-def huddle_narrow_url(realm: Realm, other_user_ids: List[int]) -> str:
+def huddle_narrow_url(*, user: UserProfile, display_recipient: List[UserDisplayRecipient]) -> str:
+    realm = user.realm
+    other_user_ids = [r["id"] for r in display_recipient if r["id"] != user.id]
```

Comment:
This would be a follow-up and should be discussed in a new #backend thread, but I seem to recall we've been trying to move towards not excluding the current user ID in group DM URLs.

Reply:
Yeah, I think that's an interesting question.  Obviously this particular change is agnostic about the decision.  It's a little funny to me that the `*_narrow_url` functions on the backend only show in email-notifications code at the moment, but it's not necessarily wrong, it was just surprising to me.